sdcm/cluster.py

```diff
@@ -1020,6 +1045,45 @@ def stop_scylla(self, verify_up=False, verify_down=True, timeout=300):
         self.stop_scylla_server(verify_up=verify_up, verify_down=verify_down, timeout=timeout)
         self.stop_scylla_jmx(verify_up=verify_up, verify_down=verify_down, timeout=timeout)
 
+    def restart_node_with_resharding(self, murmur3_partitioner_ignore_msb_bits=DEFAULT_MURMUR3_PARTITIONER_IGNORE_MSB_BITS):
+        def _wait_for_resharding(timeout=60, reshard_found=False, start_search_from_line=0):
+            """
+            wait until there's no Reshard listed in the log
+            sleep for more 5 seconds
+            break if there's no Reshard in the log
+            """
+            to = 0
+            sleep_time = 5
+            prev_out = []
+            while to <= timeout:
+                out = self.search_database_log('Reshard', start_search_from_line=start_search_from_line)
+                m = False if not out else True
+                if m and prev_out == [o[0] for o in out]:
+                    break
+                prev_out = [o[0] for o in out]
+                if not m:
+                    if not timeout:
+                        return reshard_found
+                    time.sleep(sleep_time)
+                    return _wait_for_resharding(timeout=0, reshard_found=reshard_found)
```

Comment:
I don't understand, why do you need a recursion here?

Reply:
@juliayakovlev ?

Reply:
Removed this code. Use Larisa's function